Modules/_datetimemodule.c

```diff
@@ -2998,33 +3023,32 @@ date_fromisoformat(PyObject *cls, PyObject *dtstr)
         goto invalid_string_error;
     }
 
-    return new_date_subclass_ex(year, month, day, cls);
+    return new_date_subclass_ex(year, month, day, (PyObject *) type);
 
 invalid_string_error:
     PyErr_Format(PyExc_ValueError, "Invalid isoformat string: %R", dtstr);
     return NULL;
 }
 
 
-static PyObject *
-date_fromisocalendar(PyObject *cls, PyObject *args, PyObject *kw)
-{
-    static char *keywords[] = {
-        "year", "week", "day", NULL
-    };
+/*[clinic input]
+@classmethod
+datetime.date.fromisocalendar
 
-    int year, week, day;
-    if (PyArg_ParseTupleAndKeywords(args, kw, "iii:fromisocalendar",
-                keywords,
-                &year, &week, &day) == 0) {
-        if (PyErr_ExceptionMatches(PyExc_OverflowError)) {
-            PyErr_Format(PyExc_ValueError,
-                    "ISO calendar component out of range");
+    year: int
+    week: int
+    day: int
 
-        }
```

Comment:
Now I see that before this PR we have code in `fromisocalendar()` to raise ValueError rather than OverflowError. Raising a ValueError seems like better behavior to me. Also, changing this would be considered backwards-incompatible, would require NEWS and What's New entries, and couldn't be backported.

I think we should find a way to retain this behavior, even if that means not using Argument Clinic for this method right now.